elogia_sale/models/elogia_sale.py

```diff
@@ -139,9 +139,14 @@ class CampaignAccountingSetting(models.Model):
     account_debit_second = fields.Many2one('account.account', 'Debit (Control)')
     account_credit_second = fields.Many2one('account.account', 'Credit (Control)')
 
-    @api.constrains('account_debit_first', 'account_credit_first')
+    @api.constrains('account_debit_first', 'account_credit_first', 'company_id')
     def check_control_account(self):
+        env_accounting = self.env['campaign.accounting.setting']
         for record in self:
+            if record.company_id:
+                obj_accounting = env_accounting.search([('company_id', '=', record.company_id.id), ('id', '!=', record.id)])
```

Comment:
Si el objeto 'campaign.accounting.setting' tiene el campo company_id debería tener una regla de seguridad para las compañias.
Si fuera el caso entonces en el search esto estaría de más: ('company_id', '=', record.company_id.id)

Reply:
Te refieres a esto:  <field name="domain_force">[(1,'=',1), ('company_id', 'in', company_ids)]</field>

Reply:
Para todos los usuarios:
<record model="ir.rule" id="el_id_que_quieras">
        <field name="name">El_nombre_que_quieras</field>
        <field name="model_id" ref="model_tu_modelo"/>
        <field name="global" eval="True"/>
        <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
    </record>

Reply:
Ponme un ejemplo por fa q no lo veo ahora mismo.

Reply:
<record model="ir.rule" id="royalty_parameters_comp_rule">
        <field name="name">royalty_parameters multi-company</field>
        <field name="model_id" ref="model_sale_royalty"/>
        <field name="global" eval="True"/>
        <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
    </record>